src/preact/base-element.js

```diff
@@ -647,6 +734,10 @@ function shouldMutationForNodeListBeRerendered(nodeList) {
 function shouldMutationBeRerendered(Ctor, m) {
   const {type} = m;
   if (type == 'attributes') {
+    // Check whether this is a templates attribute.
+    if (Ctor['usesTemplate'] && m.attributeName == 'template') {
```

Comment:
Is this supporting the case where `template` is identified as an attribute by the target template `id`? Let's also add a Storybook example for that.

Reply:
Correct. Will add a storybook.